src/otx/algorithms/classification/adapters/mmcls/configurer.py

```diff
@@ -284,49 +222,25 @@ def configure_loss(self, cfg):
             )
             update_or_add_custom_hook(cfg, ib_loss_hook)
 
+    def get_sampler_type(self, cfg):
+        """Return sampler type."""
+        if self._is_multiclass(cfg):
+            sampler_type = "balanced"
+        else:
+            sampler_type = "cls_incr"
+        return sampler_type
 
-class SemiSLClassificationConfigurer(ClassificationConfigurer):
-    """Patch config to support semi supervised learning for classification."""
+    def use_adaptive_repeat(self, cfg) -> bool:
+        """Return whether using adaptive repeat.
 
-    def configure_hook(self, cfg):
-        """Update cfg.custom_hooks."""
-        super().configure_hook(cfg)
-        # Set unlabeled data hook
-        if self.training:
-            if cfg.data.get("unlabeled", False) and cfg.data.unlabeled.get("otx_dataset", False):
-                self.configure_unlabeled_dataloader(cfg)
+        Currently, only multi class classification supports adaptive repeat.
+        """
+        return self._is_multiclass(cfg)
```

Comment:
Will use_adaptive_repeat always be true if it's a multi class?
What if the user has a multi-class task and doesn't want to use adaptive_repeat?

Reply:
Same with the `AdaptiveTrainingSchedule`, it will be automatically triggered. If users don't want to use, they could do by modifying recipe at the current status to remove the `CustomHook` named `AdaptiveRepeatDataHook`.

Reply:
Current only multi class classification has two methods for adaptive training. First is adaptive training with adaptive epoch validation, which is common for the other tasks. The other is adaptive training with repeat sampler. This method enlarges iterations per epoch. Currently we all agreed that default operation of adaptive training for multi class classification is repeat sampler. Therefore, if user don't want this, then user should turn off adaptive_training flag.

Reply:
But shouldn't this function always come out as True, even if the user changes adaptive repeat to disable?

Reply:
What I mean is that I don't think it's semantically correct for the use_adaptive_repeat function to return is_multiclass as it is.

Reply:
> But shouldn't this function always come out as True, even if the user changes adaptive repeat to disable?

Yes, but it does not effect training when users disable adaptive training

Reply:
> > But shouldn't this function always come out as True, even if the user changes adaptive repeat to disable?
> 
> Yes, but it does not effect training when users disable adaptive training

Yeah I was worried that from an API perspective, this is a public function and it would be confusing for users to see it always be true even when disabled.

Reply:
I agree with @jaegukhyun and @harimkang . Current implementation is still temporary since we need to more implementation for multi-label and h-label and other tasks (det., seg.). So, it'll be refactored at my later PRs.

Reply:
Yes thanks for the clarification. I guess it's a function that will be removed in the future.